### PR TITLE
Issue/#34  Used selectedId from state for adding and updating a report.

### PR DESCRIPTION
**Pull request by nggbaobkit**

### CODE REVIEW

File: training/libs/store/report/src/lib/+state/report.effects.ts [hunk 1]
```diff
@@ -83,73 +74,76 @@ export class ReportEffects {
...
+      ) => {
+        return this.reportService
+          .deleteReport(state[REPORT_FEATURE_KEY].selectedId)
+          .pipe(
             map(deletedId => {
               alert(`Deleted report with id ${deletedId}`);
               this.router.navigate(['/report']);
-              return ReportActions.deleteReportSuccess({
+              return ReportActions.deleteSelectedReportSuccess({
                 deletedId: deletedId
```
Comment (ntien97) [inline training/libs/store/report/src/lib/+state/report.effects.ts:90]: optimize code: `{ deletedId }`

File: training/libs/store/report/src/lib/+state/report.effects.ts [hunk 2]
```diff
@@ -83,73 +74,76 @@ export class ReportEffects {
...
+                  jobYesterday: action.jobYesterday,
+                  problems: action.problems,
+                  jobToday: action.jobToday
+                }
+              };
+              alert(`Updated report with id ${updatedReport.id}`);
+              this.router.navigate(['/report']);
+              return ReportActions.updateSelectedReportSuccess({
+                updatedReport: update
+              });
```
Comment (ntien97) [inline training/libs/store/report/src/lib/+state/report.effects.ts:130]:
```suggestion
            map(({id}) => {
              const updatedReport: Update<Report> = {
                id,
                changes: {
                  jobYesterday: action.jobYesterday,
                  problems: action.problems,
                  jobToday: action.jobToday
                }
              };
              alert(`Updated report with id ${id}`);
              this.router.navigate(['/report']);
              return ReportActions.updateSelectedReportSuccess({
                updatedReport
              });
```

File: training/libs/store/report/src/lib/+state/report.facade.ts [hunk 3]
```diff
@@ -26,25 +25,16 @@ export class ReportFacade {
...
-  }
-
   addReport(req: ReportCreateRequest) {
     this.dispatch(addReport({ req }));
   }
 
-  deleteReport(id: number) {
-    this.dispatch(deleteReport({ id }));
+  deleteSelectedReport() {
+    this.dispatch(deleteSelectedReport({}));
```
Comment (ntien97) [inline training/libs/store/report/src/lib/+state/report.facade.ts:33]:
```suggestion
    this.dispatch(deleteSelectedReport());
```

File: training/libs/store/report/src/lib/+state/report.facade.ts [hunk 4]
```diff
@@ -26,25 +25,16 @@ export class ReportFacade {
...
 
-  updateReport(
-    id: number,
-    jobYesterday: string,
-    problems: string,
-    jobToday: string
-  ) {
-    this.dispatch(updateReport({ id, jobYesterday, problems, jobToday }));
+  updateReport(jobYesterday: string, problems: string, jobToday: string) {
+    this.dispatch(updateSelectedReport({ jobYesterday, problems, jobToday }));
```
Comment (ntien97) [inline training/libs/store/report/src/lib/+state/report.facade.ts:37]: `{ jobYesterday, problems, jobToday }` this interface is used a lot. should you declare an interface for it & passed it through the `updateReport` function as well?

Comment (nggbaobkit) [reply training/libs/store/report/src/lib/+state/report.facade.ts:37]: This interface is actually the same as `ReportCreateRequest`. Now, should I rename `ReportCreateRequest` to `ReportCreateOrUpdateRequest` and use it for both functions addReport and updateSelectedReport or should I create two interfaces request with the exact same attributes?

Comment (ntien97) [reply training/libs/store/report/src/lib/+state/report.facade.ts:37]: Separate it. Even though the body is the same, it's used in different scenarios. In the future, if you add some attribute to the `PUT` API, you can simply update only the `ReportUpdateRequest`

**Pull request merged**